fpga_interchange/populate_chip_info.py

```diff
@@ -1740,6 +1760,426 @@ def get_cell(cell_type):
+                                if _port.instance_name is not None:
+                                    new_ports.append(_port)
+                    new_nets[net[0]] = Net(net[1].name, net[1].property_map,
+                                           new_ports)
+                if len(nets_dict) != 0:
+                    for key in nets_dict.keys():
+                        for net in nets_dict[key].items():
+                            new_nets[str(key) + "/" + str(net[0])] = net[1]
+
+            cell_instances = {}
+            nets = {}
+
+            if debug:
+                print("\tRemoving const nets")
+
+            # Remove const cells and nets that were driven by them,
+            # as they could create false root_cells
+            VCC_cell = constants.VCC_CELL_TYPE
+            GND_cell = constants.GND_CELL_TYPE
+            for net in old_nets.items():
+                add = True
+                for port in net[1].ports:
+                    temp = port.instance_name
+                    temp_name = port.name
+                    if temp is not None:
+                        temp = old_cell_instances[temp]
+                        if temp.cell_name in [VCC_cell, GND_cell]:
+                            add = False
+                    elif temp_name in macro[1].ports and\
+                         macro[1].ports[temp_name].direction != Direction.Inout:
+                        add = False
+                if add:
+                    nets[net[0]] = net[1]
+
+            for cell in old_cell_instances.items():
+                if cell[1].cell_name not in [VCC_cell, GND_cell]:
+                    cell_instances[cell[0]] = cell[1]
+
+            for cell_instance in cell_instances.items():
+                cell_name = cell_instance[1].cell_name
+                instance_to_cell_map[cell_instance[0]] = cell_name
+                graph[cell_instance[0]] = []
+            for net in nets.items():
+                sources = []
+                sinks = []
+                for port in net[1].ports:
+                    if port.instance_name is not None:
+                        cell = instance_to_cell_map[port.instance_name]
+                        if len(cell_pin_direction_map[cell]) == 0:
+                            continue
+                        direction = cell_pin_direction_map[cell][port.name]
+                        if direction:
+                            sources.append(port.instance_name)
+                        else:
+                            sinks.append(port.instance_name)
+                if sources == [] or sinks == []:
+                    continue
+
+                for source in sources:
+                    for sink in sinks:
+                        graph[source].append((sink, 1))
+                        graph[sink].append((source, 0))
+
+            cluster = {}
+            cluster['name'] = macro[0]
+            # If we can't find good root use any cell as root cell
```

Comment:
Probably this comment is out of place. Fits better when assigning the `root` a bit below in the code